### ISSUE TITLE
New command: `editor: go to diagnostic (workspace)`

### ISSUE THREAD

**Issue by DragonDev1906** [zed-industries/zed#14638]
### Check for existing issues

- [X] Completed

### Describe the feature

The `editor: go to diagnostic` (F8) command jumps to the next diagnostic in this file. However you often have the need to go to all errors regardless of which file they are in. For example when changing the type of something you want to go to all places that are affected.

There is the `diagnostics: Deploy` (`Ctrl+Shift+M`) command that shows an overview of all errors (though for some reason `F8` doesn't work there), but it only shows 5 lines which often isn't sufficient.

Would be nice to have an `editor: go to diagnostic (workspace)` command (and a corresponding `editor: go to prev diagnostic (workspace)`) which could by default be bound to `Ctrl+F8` and `Ctrl+Shift+F8`, that switches files when you've reached the last error in the currently opened file.

**Comment by JosephTLyons** [comment]
Hey @DragonDev1906, note that you can expand the the excerpts to show more context via these buttons:

<img width="203" alt="image" src="https://github.com/user-attachments/assets/1328c0d2-239a-4693-9e63-bb1ea7d83314">

There are also commands you can run and settings to tweak how many lines are revealed each time the action is ran.

**Comment by DragonDev1906** [comment]
> just wanted to note that you can expand the the excerpts to show more context via these buttons:

Yes, I have been using this as a workaround. It gives a nice overview and is really useful for fixing multiple errors that are similar, but it has other problems that make me wish for F8 to be across the entire project:
- F8 doesn't work in this diagnostics view (it doesn't jump to the next entry in the overview list)
- So far I haven't found a way to jump the cursor to the next entry (via keyboard) besides using the arrow keys or relative line number jumps (vim) (see F8 not working here)
- Go to line (ctrl+G) doesn't go to where you expect it to go: Instead of jumping to the displayed line numbers it jumps inside the diagnostics view, using line numbers that aren't displayed anywhere, thus making absolute jumps to line numbers basically useless in this view. It makes sense why it works this way, and perhaps that is even the intended/expected behavior, but it makes navigating in the diagnostics view hard.
- Excerpt expansion can get confusing when they overlap, resulting in the same line of code getting shown multiple times without any indication that this is the case until you start writing (the cursor is only in one of them, no highlighting or any other indication that this is the case except for the line numbers)

So my workflow for handling errors has often been the following:
- Go to overview
- Alt+Enter to go into the file (to see a lot more context)
- Use F8 to go through all errors in the file
- Go back to overview
- (Manually) go down to the next file with an error
- Repeat

With a project-wide F8 all of that could be replaced with a single shortcut, at least when the errors are different enough and you know they require a lot of context (e.g. when they are due to lifetimes).

**Comment by fornellas-udemy** [comment]
@JosephTLyons regarding [the arrows](https://github.com/zed-industries/zed/issues/14638#issuecomment-2246382060), it looks like they're gone:

![image](https://github.com/user-attachments/assets/4aa57309-ebcf-4813-8d0d-a3b1d2251aef)

They are still present on global find though:

![image](https://github.com/user-attachments/assets/525fdc08-4af2-41d7-949c-4aa3a81c5df6)

**Comment by bitfield** [comment]
Just to add that it's a really strange experience for new users: F8 always "goes to next diagnostic", with one glaring exception: in the diagnostics multibuffer! Here it simply does nothing at all. This seems like an easy win, unless there's some blocker I'm not seeing.

**Comment by thomaseizinger** [comment]
Alternatively, the existing `GoToDiagnostic` command could be extended with an option to also search across files. Currently, the default keymap is:

```
"f8": ["editor::GoToDiagnostic", { "severity": { "min": "hint", "max": "error" } }],
"shift-f8": ["editor::GoToPreviousDiagnostic", { "severity": { "min": "hint", "max": "error" } }],
```

This already has some configuration options so perhaps it makes more sense to add a `workspace: true` flag, or `advanced_to_next_file: true`.

Am I missing something in how other people are using zed? As far as I can see, there is no keyboard-driven workflow to just go through all the current build errors and fix them one after the other.
